CommunityShed/Pages/SearchItem.aspx.cs

```diff
@@ -0,0 +1,122 @@
+                    SELECT i.id AS itemId, i.[Name], i.CategoriesId, c.[Type], c.Id,  p.FirstName + ' ' + p.LastNAME AS FullName, ag.Age
+                    FROM Item i JOIN Categories c ON i.CategoriesId = c.Id
+                    JOIN AgeRange ag ON ag.Id = i.AgeId
+                    JOIN CommunityPerson cp ON I.CommunityPersonId = cp.Id
+                    JOIN Person p ON cp.PersonId = p.id
+                    WHERE i.[Name] = @Name AND c.Id = @toolType",
+                    new System.Data.SqlClient.SqlParameter("@name", toolName),
+                    new System.Data.SqlClient.SqlParameter("@toolType", selectedToolType)
+                    );
+                ItemsList.DataSource = dt.Rows;
+                ItemsList.DataBind();
+            }
+            //If search is conducted using only the tool type drop down list and range drop down list
+            else if (toolName.Length > 1 && range!= -1)
+            {
+                DataTable dt = DatabaseHelper.Retrieve(@"
+                    SELECT  i.id AS itemId, i.[Name], i.CategoriesId, c.[Type], c.Id,  p.FirstName + ' ' + p.LastNAME AS FullName, ag.Age
+                    FROM Item i JOIN Categories c ON i.CategoriesId = c.Id
+                    JOIN AgeRange ag ON ag.Id = i.AgeId
+                    JOIN CommunityPerson cp ON I.CommunityPersonId = cp.Id
+                    JOIN Person p ON cp.PersonId = p.id
+                    WHERE i.[Name] = @Name AND WHERE ag.Id = @selectedAgeRange",
+                    new System.Data.SqlClient.SqlParameter("@name", toolName),
+                    new System.Data.SqlClient.SqlParameter("@selectedAgeRange", range)
+                    );
+                ItemsList.DataSource = dt.Rows;
+                ItemsList.DataBind();
+            }
+            //If search is conducted using only the search textbox, tool type drop down list and range drop down list
+            else if (toolName.Length > 1 && selectedToolType != -1 && range != -1)
+            {
+                DataTable dt = DatabaseHelper.Retrieve(@"
+                    SELECT  i.id AS itemId, i.[Name], i.CategoriesId, c.[Type], c.Id,  p.FirstName + ' ' + p.LastNAME AS FullName, ag.Age
+                    FROM Item i JOIN Categories c ON i.CategoriesId = c.Id
+                    JOIN AgeRange ag ON ag.Id = i.AgeId
+                    JOIN CommunityPerson cp ON I.CommunityPersonId = cp.Id
+                    JOIN Person p ON cp.PersonId = p.id
+                    WHERE i.[Name] = @Name AND WHERE c.Id = @toolType AND WHERE ag.Id = @selectedAgeRange",
+                    new System.Data.SqlClient.SqlParameter("@name", toolName),
+                    new System.Data.SqlClient.SqlParameter("@toolType", selectedToolType),
+                    new System.Data.SqlClient.SqlParameter("@selectedAgeRange", range)
+                    );
+                ItemsList.DataSource = dt.Rows;
+                ItemsList.DataBind();
+            }
```

Comment:
I like the flexibility that you're implementing in your search feature, but this whole section could use refactoring to remove duplicated code. Happy to work through it with you when you get a moment.